basetest.pm

```diff
@@ -652,20 +652,21 @@ sub standstill_detected {
 # state
 sub rollback_activated_consoles {
     my ($self) = @_;
+    my $ret;
     for my $console (@{$self->{activated_consoles}}) {
         # the backend will only reset its state, and call activate
         # the next time - the console itself might actually not be
         # able to activate a 2nd time, but that's up to the console class
-        autotest::query_isotovideo('backend_reset_console', {testapi_console => $console});
+        $ret = autotest::query_isotovideo('backend_reset_console', {testapi_console => $console});
+        last if $ret->{error};
```

Comment:
So there's no point in trying to reset the remaining consoles anymore? This looks like it should accumulate errors but continue. If `$autotest::last_milestone_console` is defined it will actually continue to do one more query but clears a previously occurred error which seems wrong, too.

So this function is inconsistent in deciding whether to abort on the first error or whether not and possibly suppresses errors.

Reply:
So far the reset consoles call will abort the test on error. I do not want to change this but just improve the error handling and notification. Also as the test would fail anyway I do not see a point trying to continue with the other consoles. About the specific problem of "sol deactivate" the problem could very well be to catch this in the specific backend and do whatever mitigation or repair is necessary, e.g. ignore exit code, retry, etc. But here, within the proxy call we just need to ensure the error is properly propagated